### DISCUSSION TITLE
Official python library is available

### DISCUSSION THREAD

**Discussion by madpilot** [amberelectric/public-api#53] [Announcements]
I just uploaded a python library for the Amber API to PyPi: ~https://pypi.org/project/amberelectric.py/~ **See note below**

~It's mildly official in that it's not under Amber's account yet, because we are waiting for another python package (that no longer works) to release the `amberelectric` package name.~

Why a Python package? It's needed for my home assistant custom control.

I'll probably knock up a NPM package next...

**Comment by madpilot** [comment]
Update! We got the amberelectric package name! https://pypi.org/project/amberelectric/

**Reply by iaitkenpic** [reply]
Working well. Thank you

**Comment by tjhowse** [comment]
That's awesome! Thanks! I was just about to delve into writing my own in preparation for the five minute settlements. This will save me a lot of effort.

**Comment by icenov** [comment]
Just joined Amber and found the python package. Very helpfull - thanks. Working through the setup and config at [https://pypi.org/project/amberelectric/](url) I think there is an error in the get current price code:
```
current = api.get_current_prices(site_id)
```
which should be 
```
current = api.get_current_price(site_id)
```
i.e. price not prices

**Reply by iaitkenpic** [reply]
That worked. Thank you

**Reply by madpilot** [reply]
Oooh. Sorry I forgot to fix that. I'll try get to that tomorrow

**Comment by iaitkenpic** [comment]
Trying to instal this on my RPi4 running openhab3 and I get the follow:
openhabian@openhabian:~ $ pip install amberelectric
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting amberelectric
  Could not find a version that satisfies the requirement amberelectric (from versions: )
No matching distribution found for amberelectric

Not sure how to proceed

**Reply by madpilot** [reply]
What version of python are you running?

**Comment by iaitkenpic** [comment]
Hi Myles,

Thanks for your reply.

Redid the installation with 'sudo pip3 install amberelectric' and all
worked correctly.

Cheers

Ian

**Comment by iaitkenpic** [comment]
Cool, thanks!

**Comment by iaitkenpic** [comment]
Hi @madpilot, I have made some progress using your Amberelectric API. I am using Openhab3 platform to control certain devices in my home. When electricity is expensive I want to turn them off. As such I have created a rule in Openhab that executes a Python script that in turn calls your API library. This executes every 15 or so minutes. I was expecting a JSON formatted output from this but what I am getting appears to be formatted differently. Is there a way I can get a JSON output or is this my ignorance? The data I get back is: [{'duration': 30.0, 'spot_per_kwh': 3.10168, 'per_kwh': 16.77556, 'date': datetime.date(2021, 12, 7), 'nem_time': datetime.datetime(2021, 12, 7, 11, 30, tzinfo=tzoffset(None, 36000)), 'start_time': datetime.datetime(2021, 12, 7, 1, 0, 1, tzinfo=tzutc()), 'end_time': datetime.datetime(2021, 12, 7, 1, 30, tzinfo=tzutc()), 'renewables': 39.31, 'channel_type': <ChannelType.GENERAL: 'general'>, 'spike_status': <SpikeStatus.NO_SPIKE: 'none'>, 'type': 'CurrentInterval', 'range': None, 'estimate': True}]

**Reply by madpilot** [reply]
The python lib returns Python objects. You should be able to convert the back to JSON using a JSON encoder: https://docs.python.org/3/library/json.html

**Reply by madpilot** [reply]
Though it may just be easier to execute a curl command (which you can copy from the API docs) - they return JSON natively

**Reply by iaitkenpic** [reply]
Thanks @madpilot , I think curl might be the easier path to take. Thank you again for all your help
